employees/views.py

```diff
@@ -155,9 +155,23 @@ class ReportListCreateProjectJoinView(MonthNavigationMixin, ProjectsWorkPercenta
     url_name = "custom-report-list"
     object = None
 
+    def get_default_date(self) -> datetime.date:
+        year = int(self.kwargs["year"])
+        month = int(self.kwargs["month"])
+        current_date = timezone.now()
+        if current_date.year == year and current_date.month == month:
+            return current_date.date()
+        queryset = self.get_queryset()
+        if not queryset.exists():
+            return datetime.date(year=year, month=month, day=1)
+        date = queryset.first().date + timezone.timedelta(days=1)
+        if date.month != month:
+            return queryset.first().date
```

Comment:
You should evaluate queryset only once.

Reply:
I forgot this would be an issue. At this point I treated it more like list-based structure returning reference to it's head.

Is it enough to reduce `.first()` calls to one or should I maybe call `.first()` on the `get_queryset()` and call `.exists()` on that?

Reply:
Not big issue, but rather optimization. The way you did it now is OK.